django_file_form/widgets.py

```diff
@@ -78,12 +91,21 @@ def value_from_datadict(self, data, files, name):
         else:
             placeholders = get_placeholder_files(data, name)
             s3uploaded = get_s3_uploaded_files(data, name)
-            return placeholders[0] if placeholders else (s3uploaded[0] if s3uploaded else None)
+            metadata = get_file_meta(data, name)
+            obj = placeholders[0] if placeholders else (s3uploaded[0] if s3uploaded else None)
+            if obj is not None and obj.name in metadata:
+                obj.metadata = metadata[obj.name]
+            return obj
 
 class UploadMultipleWidget(UploadWidget):
     def value_from_datadict(self, data, files, name):
         if hasattr(files, 'getlist'):
-            return files.getlist(name) + get_placeholder_files(data, name) + get_s3_uploaded_files(data, name)
+            metadata = get_file_meta(data, name)
+            objs = files.getlist(name) + get_placeholder_files(data, name) + get_s3_uploaded_files(data, name)
+            for obj in objs:
+                if obj.name in metadata:
+                    obj.metadata = metadata[obj.name]
+            return objs
         else:
             # NB: django-formtools wizard uses dict instead of MultiValueDict
             return super(UploadMultipleWidget, self).value_from_datadict(data, files, name)
```

Comment:
I do not know what triggers this case and how to add `metadata` to it.

Reply:
We can skip this for now.